### ISSUE TITLE
Install dependencies of build script.

### ISSUE THREAD

**Issue by nojaf** [fsharp/FAKE#2100]
### Description

I'd like a way to download dependencies of the build script without invoking anything else.

Scenario:

I want to add a new module `nuget Fake.JavaScript.Npm`.
Before I can add `open Fake.JavaScript` I need to have that dependency installed.

### Repro steps

Please provide the steps required to reproduce the problem

1. Delete build.fsx.lock file

2. Run `fake.exe run build.fsx --help`

With `--help` no target is invoked it does contain side effects however.

### Expected behavior

I'd be nice if there were a `fake.exe install` to fetch the new dependency.

#### Option `fake install`

Similar to `paket install`. Open points: Would it write/update the intellisense file -> probably yes

#### Option `fake update`

Similar to `paket update`. Open points: Would it write/update the intellisense file -> probably yes

#### Option `fake paket <..>`

Will execute `paket <..>` (internally). And write/update intellisense files if needed?

#### Option `fake compile`

Will execute something similar to `paket install` (ie. add/download new packages), update intellisense files and compiling the script (without running it). `fake run` should be fast and cached afterwards.

This option could have additional parameters in the future:
- `fake compile --update` to update dependencies similar to deleting the lockfile and `paket update`
- `fake compile --no-install` to only compile and without `paket install`

**Comment by matthid** [comment]
After discussing what the CLI should look like (`fake install`, `fake update`, `fake prepare`, `fake download`, `fake restore` ...) I'd accept a PR for this. From time to time I thought about this as well. (you should feel free to send a PR as changing the CLI shouldn't be a problem)

**Comment by nojaf** [comment]
Hi @matthid , thanks for the approval.
Could you give me some directions on getting started?
What should the cli look like? Could it be an option to just do something like `fake.exe paket <command>` and act accordingly?

**Comment by matthid** [comment]
That would be an even bigger change and probably be seen sceptical by some users. It would need a change similar to https://github.com/fsprojects/Paket/pull/3338 before we can start here

**Comment by BlythMeister** [comment]
If I understand your aim is to perform the compile stage of running fake.  Without running your script.

So why not use a similar syntax to ’fake run <script>‘ and do ’fake compile <script>‘

**Comment by baronfel** [comment]
An additional thought I had with @theangrybyrd tonight would be to package up that compiled version of the fake dll into a single deployable artifact somehow. Ideally a .net core static-linked exe, but that might not be feasible.

**Comment by matthid** [comment]
@baronfel Yes I thought about that as well, this is useful especially for release pipelines, however this is a separate issue imho

**Comment by matthid** [comment]
@BlythMeister I like `fake compile` as well

**Comment by nojaf** [comment]
@matthid just to be on the safe side, what does `fake compile` do for you?
I'd like something that installs my fake deps. Compile sounds like you will compile that fsx to a dll?

**Comment by matthid** [comment]
@nojaf Yes it would do everything, load all deps, generate intellisense files and compile the script (cached behind the scenes), such that everything is ready for `fake run`

**Comment by matthid** [comment]
And it really would be a `fake` specific operation and could have options like `fake compile --no-install` to skip the `paket install` step if needed

**Comment by nojaf** [comment]
Ok, would you mind writing out a small summary of all the commands. What each should do?
I could perhaps start then with `fake install` and `fake update`.

**Comment by matthid** [comment]
Imho the commands are different ways to solve the problem of this issue. I'll update the first post

**Comment by matthid** [comment]
Hope the suggestions make more sense now and capture the current discussions

**Comment by matthid** [comment]
Anyone care to send a PR for this? (Currently I'd probably favor `fake compile`)

**Comment by matthid** [comment]
Related: https://github.com/fsharp/FAKE/issues/1783
